### PR TITLE
Handle null values in ingested json metadata

### LINKED ISSUE #709: Problem: metadata.json ingest breaks if null values are present
When ingesting a transfer with a metadata.json file, Archivematica converts the json to csv.   If any of the values in the json are null, the converstion to csv fails, halting processing of that transfer.

The error that shows up in the logs is generated by:

https://github.com/artefactual/archivematica/blob/stable/1.6.x/src/MCPClient/lib/clientScripts/jsonMetadataToCSV.py#L71

and looks like:
`
DEBUG     2017-07-31 12:56:01  archivematica.mcp.server:taskStandard:check_request_status:95:  Task a86940c5-250d-4982-be53-5cbf3abf3e95 finished! Result COMPLETE - {'stdOut': '', 'exitCode': 1, 'stdError': 'Traceback (most recent call last):\n  File "/usr/lib/archivematica/MCPClient/clientScripts/jsonMetadataToCSV.py", line 129, in <module>\n    sys.exit(main(sip_uuid, json_metadata))\n  File "/usr/lib/archivematica/MCPClient/clientScripts/jsonMetadataToCSV.py", line 119, in main\n    writer.writerow(object_to_row(fix_encoding(row), headers))\n  File "/usr/lib/archivematica/MCPClient/clientScripts/jsonMetadataToCSV.py", line 80, in fix_encoding\n    return {key.encode(\'utf-8\'): encode_item(value) for key, value in row.items()}\n  File "/usr/lib/archivematica/MCPClient/clientScripts/jsonMetadataToCSV.py", line 80, in <dictcomp>\n    return {key.encode(\'utf-8\'): encode_item(value) for key, value in row.items()}\n  File "/usr/lib/archivematica/MCPClient/clientScripts/jsonMetadataToCSV.py", line 71, in encode_item\n    return [i.encode(\'utf-8\') for i in item]\nAttributeError: \'NoneType\' object has no attribute \'encode\'\n'}`

It should be possible to continue processing, ignoring the null value.

### PR DESCRIPTION
**Pull request by thatandromeda**
Resolves #709.

(I also owe you a contributor agreement, which I need to clear with work, though I don't anticipate any problems.)

### CODE REVIEW

File: src/MCPClient/lib/clientScripts/jsonMetadataToCSV.py [hunk 1]
```diff
@@ -65,7 +65,9 @@ def encode_item(item):
     """
     Wraps str.encode by recursively encoding lists.
     """
-    if isinstance(item, basestring):
+    if not item:  # Handle case where json contains null.
+        return
+    elif isinstance(item, basestring):
         return item.encode('utf-8')
     else:
         return [i.encode('utf-8') for i in item]
```
Comment (jraddaoui) [inline src/MCPClient/lib/clientScripts/jsonMetadataToCSV.py]: I think we should handle the same case inside this `for` loop, for `null` values inside lists. For example:

```
[
  {
    "dc.identifier": "foo",
    "filename": "objects/foo.jpg",
    "dc.subject": [
      "bar",
      null
    ]
  }
]
```

File: src/MCPClient/tests/test_json_conversion.py [hunk 2]
```diff
@@ -7,8 +7,12 @@
 JSON_MULTICOLUMN = '[{"filename": "objects/test.txt", "dc.subject": ["foo", "bar", "baz"], "dc.title": "This is a test item"}]'
 CSV_MULTICOLUMN = 'filename,dc.subject,dc.subject,dc.subject,dc.title\r\nobjects/test.txt,foo,bar,baz,This is a test item\r\n'
 
+JSON_NULL = '[{"dc.title": "This is a test item", "filename": "objects/test.txt", "spurious metadata": null}]'
+CSV_NULL = 'filename,dc.title,spurious metadata\r\nobjects/test.txt,This is a test item,\r\n'
+
```
Comment (jraddaoui) [inline src/MCPClient/tests/test_json_conversion.py:12]: If you could update or add a new test case with a `null` value inside a list, that would be great.

**Review by jraddaoui — CHANGES_REQUESTED** [review]
Thank you @thatandromeda, this is looking great!

See my comments bellow.

**Comment by thatandromeda** [comment]
OK, I've added the changes, and I've also emailed my CLA to agreement@artefactual.

**Comment by sevein** [comment]
Do any of you know if there is a reason why these tests are using `subprocess`? It seems very inefficient right?

**Comment by thatandromeda** [comment]
I don't - I was copy/pasting existing tests.

**Comment by sevein** [comment]
@thatandromeda hey did you target the stable/1.6.x because you're hoping your fix to be included in AM 1.6.2? Generally we target `qa/1.x`, for the next minor release.

@jhsimpson thoughts?

**Comment by thatandromeda** [comment]
I used whatever default Github handed me. Using Github to rebase this looks messy but I can try to rebase it locally - you want me to have started on, and be planning to merge into, `qa/1.x`?

**Comment by sevein** [comment]
I'm hoping @jhsimpson can make that call. I've heard he's thinking on a AM 1.6.2 release but I haven't really been in the loop. Merging into 1.6 and cherry-picking (we can do that) would be ok to me too.

Thank you!

**Comment by jraddaoui** [comment]
Thank you @thatandromeda, I'll give it a try today before merging.

@sevein, I chatted about this yesterday with @jhsimpson and this should go to stable/1.6.x too, so we could merge in here and cherry-pick to qa/1.x. However, I'm wondering what should be the best way to do it, should I ask @thatandromeda to squash the three commits? I think it will change the author if I do it myself. Or should I merge (--ff-only) the three commits directly? Thanks!

**Comment by thatandromeda** [comment]
Awaiting your preferences on git history - whatever you want is fine.

**Comment by jraddaoui** [comment]
Thanks @thatandromeda! Just tested it and works as expected.

If you could squash the last two commits in one and make sure your stable/1.6.x branch is up-to-date with this one, that would be perfect ;)

**Comment by jraddaoui** [comment]
Thank you again @thatandromeda!

I've added https://github.com/artefactual/archivematica/commit/4f66cf004fbc7a27d572ac2e73350d56461ea0b7 to qa/1.x too ;)
